src/backend/distributed/utils/colocation_utils.c

```diff
@@ -0,0 +1,189 @@
+/*-------------------------------------------------------------------------
+ *
+ * colocation_utils.c
+ *
+ * This file contains functions to perform useful operations on co-located tables.
+ *
+ * Copyright (c) 2014-2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/genam.h"
+#include "access/heapam.h"
+#include "access/htup_details.h"
+#include "distributed/colocation_utils.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/multi_logical_planner.h"
+#include "utils/rel.h"
+
+
+static List * DistTableOidList(void);
```

Comment:
As we talked in person, it'd be great to have regression tests for these low-level functions.

If you check `src/test` folder you can find useful examples of doing that. My favourite test there is `prune_shard_list.c`